sakura_core/outline/CDlgFuncList.cpp

```diff
@@ -600,7 +600,7 @@ void CDlgFuncList::SetData()
 			ListView_SetItem( hwndList, &item);
 
 			item.mask = LVIF_TEXT;
-			item.pszText = const_cast<WCHAR*>(pcFuncInfo->m_cmemFuncName.GetStringPtr());
+			item.pszText = const_cast<WCHAR*>(pcFuncInfo->m_cmemFuncName.c_str());
```

Comment:
分かっててやってると思いますが、一般的に const_cast の使用は不適切です。
```suggestion
			item.pszText = pcFuncInfo->m_cmemFuncName.data();
```
としたらイケる場合があります（const std::wstringな場合は無理ですが。）

Reply:
> 分かっててやってると思いますが、一般的に const_cast の使用は不適切です。

まず元のコードで `const_cast` 使ってるのでそれを何も考えずに踏襲しています。あと代入先の `LVITEMW::pszText` の型が `LPWSTR` (`WCHAR*`) なのでキャストが入るのは仕方が無い気がします。

berryzplusさんがどういう考えでこういう事を書いているかというのは自分にはよくわかりません。C++ではconst修飾子を付ける事で型システムがコンパイル時に検査してくれるから、それから逸脱するコードを限りなく避けるべきだという考えですか？自分はサクラエディタの既存のコードを使う延長線上でそこまで厳密で安全なプログラミングを追求したいわけではないです。表面上の `const_cast` を無くす事で何がそんなにおいしい結果に繋がるのかさっぱりわかりません。

あと `data` メソッドを使ってこのケースでは無理な事を確認しました。`const CFuncInfo*	pcFuncInfo;` だからです。

Reply:
windows APIの型定義が原因で発生するキャストですよね。必要があるからキャストしてるわけなので、キャスト自体に問題はないと思っています。「合法」という言い方は嫌いなんですが、合法なキャストです。

ここで指摘したかったのは、不適切なキャストをせざるを得ない実装をあちこち書くの？ということで、「自称」StdApi.cppなどの独自定義APIラッパーを作成したほうがいいんじゃない？ということでした。

Reply:
> ここで指摘したかったのは、不適切なキャストをせざるを得ない実装をあちこち書くの？ということで、「自称」StdApi.cppなどの独自定義APIラッパーを作成したほうがいいんじゃない？ということでした。

`一般的に const_cast の使用は不適切です。` というコメントの真意が `独自定義APIラッパーを作成したほうがいいんじゃない？` だったんですか。それは随分と遠回しに感じますがご出身は京都ですか？

煩雑な記述を隔離してすっきりさせたいという気持ちは分かりますが、もしそういう事をするなら別のPRで全体的に対処するやり方でないと片付かない気がします。

StdApi.cppが独自定義APIラッパーなのかといわれて見てみましたが内容を見てもしっくりきません。CommonControl.h や StdControl.h の方であればまぁラッパーぽい気がします。

あとこの件に関して調べてみましたが `ListView_SetItemText` マクロを使う事でコードの記述を少しすっきりさせる事は出来ると思います。ただPRの差分量は増えますし（既に多いのでもはや野となれ山となれかもしれませんが）ここの記述だけ更新してもサクラエディタのソースコード全体に対して対処するわけではないので統一感が出ないと思います。